server/graphql/v2/object/Host.js

```diff
@@ -427,6 +442,113 @@ export const Host = new GraphQLObjectType({
           };
         },
       },
+      contributionStats: {
+        type: new GraphQLNonNull(ContributionStats),
+        args: {
+          accounts: {
+            type: new GraphQLList(new GraphQLNonNull(AccountReferenceInput)),
+            description: 'A collection of accounts for which the contribution stats should be returned.',
+          },
+          from: {
+            type: GraphQLString,
+            description: "Inferior date limit in which we're calculating the contribution stats",
+          },
+          to: {
+            type: GraphQLString,
+            description: "Superior date limit in which we're calculating the contribution stats",
+          },
+        },
+        async resolve(host, args, req) {
+          if (!req.remoteUser?.isAdmin(host.id)) {
+            throw new Unauthorized('You need to be logged in as an admin of the host to see the contribution stats.');
+          }
+          const where = { HostCollectiveId: host.id, kind: 'CONTRIBUTION' };
+          const dateRange = dateRangeHelper(args.from, args.to);
+          if (dateRange) {
+            where.createdAt = dateRange;
+          }
+          if (args.accounts) {
+            const collectiveIds = args.accounts.map(account => account.legacyId);
+            where.CollectiveId = { [Op.in]: collectiveIds };
+          }
+          const contributions = await models.Transaction.findAndCountAll({
+            where,
+          });
+          const numOneTime = await models.Transaction.count({
+            where,
+            include: [{ model: models.Order, where: { interval: null } }],
+          });
+          const numRecurring = await models.Transaction.count({
+            where,
+            include: [{ model: models.Order, where: { interval: { [Op.ne]: null }, status: 'ACTIVE' } }],
+          });
+          const dailyAvgIncome =
+            contributions.rows.map(contribution => contribution.amount).reduce((result, item) => result + item, 0) /
+              contributions.count || 0;
+          return {
+            numContributions: contributions.count,
+            numOneTime,
+            numRecurring,
+            dailyAvgIncome,
```

Comment:
Sounds like we could rely on GraphQL features to run the queries asynchronously:

```suggestion
            numContributions: models.Transaction.count(...),
            numOneTime: models.Transaction.count(...),
            numRecurring: models.Transaction.count(...),
            dailyAvgIncome: models.Transaction.count(...),
```

Reply:
@Betree Should they be in separate resolvers?

Reply:
That's an option, I don't have any strong opinion at this point since grouping everything in the same resolvers could also allow for some nice optimizations. For example, stuff like:

```es6
const stats = models.Transaction.({ ..., groupBy: 'order.interval' });
return {
  ...,
  numOneTime: stats.find(s => !s.interval).count,
  numRecurring: stats.find(s => s.interval).count,
}
```

Reply:
For this one, I've changed the queries to run asynchronously except the queries related to calculating the averages; since both the count and sum needs to be calculated before-handed. Do you think that's acceptable, or are there any further optimizations to be done? 🤔